gateway/api/management/commands/update_jobs_statuses.py

```diff
@@ -36,15 +38,34 @@ def handle(self, *args, **options):
                 else:
                     success = False
 
-                if not success:
-                    kill_ray_cluster(job.compute_resource.title)
-                    job.compute_resource.delete()
-                    job.compute_resource = None
-                    job_status = Job.FAILED
-                    job.logs = (
-                        f"{job.logs}\nSomething went wrong during updating job status."
+                timeout = config.PROGRAM_TIMEOUT
+                if job.updated:
+                    endtime = job.updated + timedelta(days=timeout)
+                else:
+                    endtime = job.created + timedelta(days=timeout)
+                now = datetime.now(tz=endtime.tzinfo)
+                if endtime < now:
+                    job_status = Job.STOPPED
+                    logger.warning(
+                        "Job [%s] reached maximum runtime [%s] days and stopped.",
+                        job.id,
+                        timeout,
                     )
 
+                if not success:
```

Comment:
we do not need to touch resources at all in this function. Just setting job to stop status would be enough. `free_resources` will handle cluster deletion automatically

Reply:
a.k.a we can remove those additional lines :) 

and also it would be good to add
```python
job.logs = f"{job.logs}.\nMaximum job runtime reached. Stopping the job."
```

Reply:
The kill_ray_cluster has been in this function.  I've just added code not to kill it when the  "RAY_CLUSTER_NO_DELETE_ON_COMPLETE" is true.  I can take out this block.

Reply:
oh, I see. Hmm, I will check it in a little bit. But overall looks great! :) We ill merge it